### PR TITLE
[DevSpaces]: Add hostEndpointType to controller resource

### PR DESCRIPTION
**Pull request by rakeshvanga**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/devspaces/resource-manager/Microsoft.DevSpaces/stable/2019-04-01/devspaces.json [hunk 1]
```diff
@@ -578,6 +578,20 @@
...
         },
+        "hostEndpointType": {
+          "description": "The endpoint type to be exposed for a Azure Dev Spaces Controller.",
+          "enum": [
+            "Public",
+            "Private",
+            "None"
+          ],
+          "type": "string",
+          "readOnly": true,
```
Comment (sanar-microsoft) [inline specification/devspaces/resource-manager/Microsoft.DevSpaces/stable/2019-04-01/devspaces.json]: This is not supposed to be readonly, right?

**Review by sanar-microsoft — APPROVED** [review]
LGTM

**Comment by rakeshvanga** [comment]
@NullMDR, can you re-run the azure pipeline. it doesn't seem to be related to my changes.

**Comment by NullMDR** [comment]
@rakeshvanga Back to green now

**Comment by rakeshvanga** [comment]
@NullMDR, Thanks for re-running the pipeline. I would wait for the ARM to approve and merge the changes.

**Comment by rakeshvanga** [comment]
Ping...
Can some one look into merging this change?

File: specification/devspaces/resource-manager/Microsoft.DevSpaces/stable/2019-04-01/devspaces.json [hunk 2]
```diff
@@ -578,6 +578,19 @@
           "type": "string",
           "readOnly": true
         },
+        "hostEndpointType": {
```
Comment (pilor) [inline specification/devspaces/resource-manager/Microsoft.DevSpaces/stable/2019-04-01/devspaces.json]: >"hostEndpointType" [](start = 8, length = 18)

An addition of a mutable property should involve a new api-version.  See the "breaking changes" doc here: [https://armwiki.azurewebsites.net/api_contracts/Intro.html](https://armwiki.azurewebsites.net/api_contracts/Intro.html)

**Review by pilor — CHANGES_REQUESTED** [review]
1 comment

**Comment by rakeshvanga** [comment]
Would update the version for arm spec in a different PR.

**Pull request closed without merging**